common/flatpak-run-wayland.c

```diff
@@ -179,15 +179,19 @@ flatpak_run_add_wayland_args (FlatpakBwrap *bwrap,
   g_autofree char *sandbox_wayland_socket = NULL;
   gboolean res = FALSE;
   struct stat statbuf;
+  G_GNUC_UNUSED gboolean security_context_available = FALSE;
 
 #ifdef ENABLE_WAYLAND_SECURITY_CONTEXT
-  gboolean security_context_available = FALSE;
-  if (flatpak_run_add_wayland_security_context_args (bwrap, app_id, instance_id,
-                                                     &security_context_available))
-    return TRUE;
-  /* If security-context is available but we failed to set it up, bail out */
-  if (security_context_available)
-    return FALSE;
+  // WAYLAND_SOCKET is not compatible with security context.
+  if (getenv("WAYLAND_SOCKET") == NULL)
```

Comment:
I think what we really want here is to make sure that `WAYLAND_SOCKET` is valid and in that case have an early exist. All the other setup code below is not applicable in that case.

Reply:
This can even be moved above the `#ifdef ENABLE_WAYLAND_SECURITY_CONTEXT`.